docs/redesign-proposal.md

```diff
@@ -0,0 +1,93 @@
+event one is required for the proper functioning of that application.
+
+Each Pilot should also be able to expose a metrics endpoint to be scraped by
+systems like Prometheus. This could be implemented via some of the existing
+exporters for common applications, or directly within the Pilot itself if
+required.
+
+## navigator-controller-manager
+
+The `navigator-controller-manager` should implement all of the control features
+that are required for the proper functioning of the system. It should behave
+very much like the `kube-controller-manager`. If, for example, a user creates
+an `ElasticsearchCluster` resource, the `navigator-controller-manager` will
+create whatever corresponding resources are required in the target orchestrator
+in order to fulfill the request. It should then report status information of
+its actions back to the `navigator-apiserver`. Additional status information
+may also be provided by the Pilots themselves in this case, and in fact the
+`controller-manager` may also consequently perform *other* actions as a result.
+
+If a user were to modify the size of a node pool within an
+ElasticsearchCluster, the controller-manager would consequently notify the
+relevant Pilots of this event. It could then wait until a successful data
+migration has occured before actually scaling down the appropriate
+StatefulSet/Deployment. The precise semantics of *how* the controller-manager
+should communicate this to the Pilots is undefined, but however it is done it
+should be via the API server (either by updating the `status` field of the
+ElasticsearchCluster resource, or by creating an additional `ScaleEvent`
+resource to be fulfilled by the Pilots).
+
+Only one instance of the controller-manager should run at a given time, and
+they should leader-elect amongst themselves using either the
+`navigator-apiserver` or some other leader election primitive provided by the
+target orchestrator.
+
+## navigator-apiserver
+
+The apiserver component acts as the primary API surface and data storage for
+all resources types that we define as part of our API. It's design should be
+heavily modelled upon that of `kube-apiserver`, in that it should not perform
+much actual control logic itself.
+
+The `navigator-apiserver` will understand multiple versions of our types, thus
+allowing us to support `v1` configuration for types even with the newest
+versions of the `navigator-controller-manager` by losslessly converting between
+the old and new version types. More information about this can be found in
+my talk on API internals, as well as on the upcoming blog post.
+
+Like the `kube-apiserver`, multiple instances of it may run at one time to
+provide scalability on both reads and writes. The underlying data store may be
+`etcd`, Kubernetes CRDs or OpenShift TPRs. It is undecided which will be
+implemented first, however good support for `etcd` is provided in the
+`k8s.io/apiserver` library.
```

Comment:
If not custom resources or third party resource, will users  be able to use `kubectl` to interact with it?